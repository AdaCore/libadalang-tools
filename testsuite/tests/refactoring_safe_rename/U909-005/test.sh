# Rename the parameter on an interface
safe_rename -P default.gpr -S foo.ads -L 3 -R 53 -N PP --algorithm=analyse_ast
# Rename the parameter on the spec of an interface implementation
safe_rename -P default.gpr -S foo.ads -L 5 -R 64 -N PP --algorithm=analyse_ast
# Rename the parameter on the body of an interface implementation
safe_rename -P default.gpr -S foo.adb -L 2 -R 64 -N PP --algorithm=analyse_ast
