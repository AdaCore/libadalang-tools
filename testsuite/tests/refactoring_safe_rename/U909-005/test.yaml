description: |
    This test checks if the safe rename tool correctly handles
    the rename of parameters of an access to a subprogram.

    Test output in incomplete. P_Find_All_References on a parameter of an access
    to a subprogram does not find the canonical part. This will be fixed in
    U909-017.

driver: shell_script
