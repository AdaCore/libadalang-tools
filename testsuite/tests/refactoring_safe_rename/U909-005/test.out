# Renaming node foo.ads:3:53: 
References:
Unit:foo.ads
Line: 3
Source Location Range:
Start_Line: 3
Start_Column: 53
End_Line: 3
End_Column: 54

# Renaming node foo.ads:5:64: 
References:
Unit:foo.ads
Line: 5
Source Location Range:
Start_Line: 5
Start_Column: 64
End_Line: 5
End_Column: 65

# Renaming node foo.adb:2:64: 
References:
Unit:foo.adb
Line: 2
Source Location Range:
Start_Line: 2
Start_Column: 64
End_Line: 2
End_Column: 65
Line: 6
Source Location Range:
Start_Line: 6
Start_Column: 12
End_Line: 6
End_Column: 13

